src/assessments/seo/keyphraseLengthAssessment.js

```diff
@@ -1,31 +1,116 @@
-var AssessmentResult = require( "../../values/AssessmentResult.js" );
+let Assessment = require( "../../assessment.js" );
+let AssessmentResult = require( "../../values/AssessmentResult.js" );
+const inRange = require( "lodash/inRange" );
+const merge = require( "lodash/merge" );
 
 /**
- * Assesses the keyphrase presence and length
- *
- * @param {Paper} paper The paper to use for the assessment.
- * @param {Researcher} researcher The researcher used for calling research.
- * @param {Jed} i18n The object used for translations
- * @returns {AssessmentResult} The result of this assessment
-*/
-function keyphraseAssessment( paper, researcher, i18n ) {
-	var keyphraseLength = researcher.getResearch( "keyphraseLength" );
-
-	var assessmentResult = new AssessmentResult();
-
-	if ( ! paper.hasKeyword() ) {
-		assessmentResult.setScore( -999 );
-		assessmentResult.setText( i18n.dgettext( "js-text-analysis", "No focus keyword was set for this page. " +
-			"If you do not set a focus keyword, no score can be calculated." ) );
-	} else if ( keyphraseLength > 10 ) {
-		assessmentResult.setScore( 0 );
-		assessmentResult.setText( i18n.dgettext( "js-text-analysis", "The keyphrase is over 10 words, a keyphrase should be shorter." ) );
+ * Assessment to check whether the keyphrase has a good length.
+ */
+class KeyphraseLengthAssessment extends Assessment {
+	/**
+	 * Sets the identifier and the config.
+	 *
+	 * @param {Object} config The configuration to use.
+	 *
+	 * @returns {void}
+	 */
+	constructor( config = {} ) {
+		super();
+
+		let defaultConfig = {
+			parameters: {
+				recommendedMinimum: 1,
+				recommendedMaximum: 4,
+				acceptableMaximum: 8,
+			},
+			good: {
+				score: 9,
+				resultText: "Your keyphrase has a nice length.",
+				requiresLengthAndMax: false,
+			},
+			okay: {
+				score: 6,
+				resultText: "Your keyphrase is %1$d words long. That's more than the recommended maximum of %2$d words. " +
```

Comment:
Translator comments should be included, see other assessments with variables. I think for this it might be necessary to return the individual resultTexts in `i18n.dngettext` functions.